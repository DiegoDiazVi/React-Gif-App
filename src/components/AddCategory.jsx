import React, { Fragment, useState } from 'react';

export const AddCategory = ({onNewCategory, onDeleteCategories}) => {
    //hook
    const [inputValue, setInputValue] = useState('');
    //evento para setear el input
    const onInputChange = (e) =>{
        //e.target.value saca el valor que escribimos
        setInputValue(e.target.value);
    }
    //Evento para el submit
    const onSubmit = (e) => {
        //previene el comportamiento por defecto 
        e.preventDefault();
        //valida que el valor ingresado al menos tenga mas de 1 caracter, de lo contrario no lo agrega
        const cleanInpuntValue = inputValue.trim();
        if (cleanInpuntValue.length<=1) {
            return;
        }
        onNewCategory(cleanInpuntValue);
        setInputValue('');
    }
    //Evento que elimina todas las imagenes
    const onDelete = () => onDeleteCategories();
    
    return (
        <Fragment>
            <form className='form-container' onSubmit={ onSubmit }>
                <input 
                    type="text"
                    placeholder = 'Buscar gifs' 
                    value={inputValue}
                    onChange = { onInputChange }
                />
            </form>
            <div className='button-container'>
                <button onClick={onDelete}>Limpiar</button>
            </div>
        </Fragment>
    )
}
