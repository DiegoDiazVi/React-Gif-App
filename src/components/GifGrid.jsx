import React, { Fragment } from 'react'
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { PropTypes } from 'prop-types';

//recibe la propiedad categoria o el nombre del input
export const GifGrid = ({category}) => {

    /* Uso del Hook creado que pretende retornar un objeto con el array de imagenes y el bool de isLoading 
    y se le envia al hook como parametro la categoria */
    const {images, isLoading} = useFetchGifs( category );

    return (
        <Fragment>
            {/* Nombre de la categoria */}
            <h3>{category}</h3>
            {
                isLoading && <h2>Cargando gifs</h2>
            }
            {/* Importante para nombrar clases no se usa class='clase' sino className='clase' */}
            <div className='card-grid'>
                {
                    // IteraciÃ³n del array images para renderizar
                    images.map( (image) => {
                        return (
                            // Se llama el nuevo componente GifGridItem y se envia propiedades
                            <GifGridItem 
                                // Se agrega el keu que es el id de la imagen
                                key={image.id}
                                // De esta manera enviamos todas las propiedades de image
                                {...image}
                            />
                        ) 
                    })
                }
            </div>
        </Fragment>
    )
}

GifGrid.propTypes = {
    category : PropTypes.string.isRequired
}