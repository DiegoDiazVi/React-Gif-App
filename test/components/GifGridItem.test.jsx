import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../src/components/GifGridItem";

describe('Test a GifGridItem', () => {
    const title = 'Soy un gif';
    const url   = 'https://gif.com/titulo.jpg';

    test('Debe de hacer match con el snapshot', () => {
        const { container } = render( <GifGridItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    });

    test('Debe de mostrar la imagen con la url y alt indicado', () => {
        render( <GifGridItem title={title} url={url}/>);
        const {src, alt} = screen.getByRole('img')
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Debe de mostrar el titulo en el componente', () => {
        render( <GifGridItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
    });
});