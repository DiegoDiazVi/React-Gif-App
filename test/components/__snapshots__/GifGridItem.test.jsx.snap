// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test a GifGridItem Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Soy un gif"
      src="https://gif.com/titulo.jpg"
    />
    <p>
      Soy un gif
    </p>
  </div>
</div>
`;
